proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/metadata/MetadataCertsPublishingResource.java

```diff
@@ -24,40 +25,49 @@
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
-import java.util.Base64;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 import static java.text.MessageFormat.format;
 
+@Path("/")
 public class MetadataCertsPublishingResource {
 
     private static final String BEGIN_LINE = "-----BEGIN CERTIFICATE-----";
     private static final String END_LINE = "-----END CERTIFICATE-----";
     private static final String COMMON_NAME_PREFIX = "CN=";
     private static final DateFormat DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy");
 
+    private static MetadataSigningCertsView metadataSigningCertsView;
+
     @Context
     private UriInfo uriInfo;
 
     private URI metadataPublishPath;
     private URI metadataCACertsFilePath;
-    private String metadataSigningCertBase64;
+    private URI metadataSigningCertFilePath;
 
     @Inject
     public MetadataCertsPublishingResource(
-            @Named("metadataSigningCertBase64") String metadataSigningCertBase64,
+            @Named("metadataSigningCertFilePath") URI metadataSigningCertFilePath,
             @Named("metadataCACertsFilePath") URI metadataCACertsFilePath,
             @Named("metadataPublishPath") URI metadataPublishPath) {
-        this.metadataSigningCertBase64 = metadataSigningCertBase64;
+        this.metadataSigningCertFilePath = metadataSigningCertFilePath;
         this.metadataCACertsFilePath = metadataCACertsFilePath;
         this.metadataPublishPath = metadataPublishPath;
     }
 
     @GET
     public MetadataSigningCertsView getMetadataSigningCerts() {
+        if (metadataSigningCertsView == null) {
+            metadataSigningCertsView = generateMetadataSigningCertsView();
```

Comment:
If the view is cached in this way it tightly couples the rotating of the certs with a release of the app. Is this what we want?

Reply:
I think we always have to release the apps to rotate the certs - so I think this is okay

Reply:
But we can revisit this in the future when we implement the rotation process